lib/helpers/url-utils.js

```diff
@@ -29,8 +29,15 @@ function appendQueryParam(url, key, value) {
   }
   var punctuation = /\?/.test(url) ? '&' : '?';
   var query = encodeURIComponent(key);
-  if (value !== undefined && value !== '' && value !== true) {
+  if (
+    value !== undefined &&
+    value !== '' &&
+    value !== true &&
+    key !== 'setfilter'
+  ) {
     query += '=' + encodeValue(value);
+  } else {
+    query += '=' + value;
```

Comment:
Looks like this code will preserve `setfilter` value so it doesn't get encoded. Right now `addlayer` is being encoded. The encoded versions and decoded versions both work. 

We should decide if both values should be encoded or left decoded to make it consistent. It seems like the default is to *encode* which I'm assuming is how all other APIs in this library are being handled by this function - @danswick do you see any reason why we shouldn't stick with encoding for these values? 

I think the decoded values are much easier to read and take up fewer characters. Although encoding characters seems to be a good practice, plus the developer could wrap `decodeURIComponent` around `url()` if they desired a prettier URL (which is what I'm currently doing as I'm developing with these new params for the static playground).

My vote leans towards encoding.


Examples:

`addlayer` - encoded (current state):

```
https://api.mapbox.com/styles/v1/mapbox/light-v10/static/-77.03968,38.89744,5/300x200?addlayer=%7B%22id%22%3A%22road-overlay%22%2C%22type%22%3A%22line%22%2C%22source%22%3A%22composite%22%2C%22source-layer%22%3A%22road%22%2C%22filter%22%3A%5B%22%3D%3D%22%2C%5B%22get%22%2C%22class%22%5D%2C%22motorway%22%5D%2C%22paint%22%3A%7B%22line-color%22%3A%22%23ff0000%22%2C%22line-width%22%3A5%7D%7D&access_token=
```

`addlayer` - decoded:
```
https://api.mapbox.com/styles/v1/mapbox/light-v10/static/-77.03968,38.89744,5/300x200?addlayer={"id":"road-overlay","type":"line","source":"composite","source-layer":"road","filter":["==",["get","class"],"motorway"],"paint":{"line-color":"%23ff0000","line-width":5}}&access_token=
```

`setfilter` - encoded: 

```
https://api.mapbox.com/styles/v1/mapbox/streets-v11/static/-91,60,2/800x600?setfilter=%5B%22%3D%3D%22%2C%22name_en%22%2C%22Canada%22%5D&layer_id=country-label&access_token=
```

`setfilter` - decoded: 

```
https://api.mapbox.com/styles/v1/mapbox/streets-v11/static/-91,60,2/800x600?setfilter=["==","name_en","Canada"]&layer_id=country-label&access_token=
```

Reply:
Really good point! i also lean toward encoding. Coincidentally, I had a dream about this overnight and realized that all of the changes I made yesterday are unnecessary 😂. I reverted all of my URL utils changes and changed the static service to stringify `setFilter` values instead, which I think solves both problems https://github.com/mapbox/mapbox-sdk-js/pull/330/files#diff-a033a8610015f7c46139070fceb2ff39R235.